src/proto/h1/dispatch.rs

```diff
@@ -623,17 +611,8 @@ cfg_client! {
             }
         }
 
-        fn poll_ready(&mut self, cx: &mut task::Context<'_>) -> Poll<Result<(), ()>> {
-            match self.callback {
-                Some(ref mut cb) => match cb.poll_canceled(cx) {
-                    Poll::Ready(()) => {
-                        trace!("callback receiver has dropped");
-                        Poll::Ready(Err(()))
-                    }
-                    Poll::Pending => Poll::Ready(Ok(())),
-                },
-                None => Poll::Ready(Err(())),
-            }
+        fn poll_ready(&mut self, _cx: &mut task::Context<'_>) -> Poll<Result<(), ()>> {
+            Poll::Ready(Ok(()))
```

Comment:
I believe for the client the original stuff should stay. It's how we know that the user gave up on a `client.request(req).await` future, and so _this_ task should shutdown (if the callback receiver is dropped).

Reply:
Good point, I reverted the code “from memory” and forgot that `Client` also had some additional logic here.

Done.